### PR TITLE
Add createCollection API to the SDK

### PR DESCRIPTION
**Pull request by heliuchuan**
### Description
This adds a function for users to invoke 0x4::aptos_token::create_collection via the SDK

### Test Plan

pnpm test

### CODE REVIEW

**Review by 0xmaayan — DISMISSED** [review]
this should be under `digitalAsset ` class as all `aptos_token` operations. we can skip it for now as this is a small implementation of a bigger item.
Or at least have this api to be under digitalAsset

**Comment by gregnazario** [comment]
> this should be under `digitalAsset ` class as all `aptos_token` operations. we can skip it for now as this is a small implementation of a bigger item. Or at least have this api to be under digitalAsset

Yes, let's combine `collection` and `token` under `digitalAsset`

**Comment by gregnazario** [comment]
@heliuchuan just for reference, I addressed the merge conflicts, we can get these two PRs in, and then combine them into one digitalAssets, which should just be an easy rename.

File: tests/e2e/api/collection.test.ts [hunk 1]
```diff
@@ -1,44 +1,54 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-// import { Aptos, AptosConfig, Network } from "../../../src";
+import { Account, Aptos, AptosConfig, Network } from "../../../src";
+import { waitForTransaction } from "../../../src/internal/transaction";
+import { FUND_AMOUNT } from "../../unit/helper";
+
+// use it here since all tests use the same configuration
+const config = new AptosConfig({ network: Network.DEVNET });
```
Comment (gregnazario) [inline tests/e2e/api/collection.test.ts:9]: I believe we will want to change these to LOCAL, but let's evaluate that at a combined level

**Review by gregnazario — APPROVED** [review]

File: src/api/collection.ts [hunk 2]
```diff
@@ -15,6 +22,43 @@ export class Collection {
...
+      creator: Account;
+      description: string;
+      name: string;
+      uri: string;
+      options?: GenerateTransactionOptions;
+    } & CreateCollectionOptions,
+  ): Promise<SingleSignerTransaction> {
+    return createCollectionTransaction({ aptosConfig: this.config, ...args });
+  }
+
```
Comment (gregnazario) [inline src/api/collection.ts:61]: Returning a SingleSigner may be weird if we also support Fee payer, but I'm not sure how that model is going to look cc @0xmaayan

**Pull request merged**